react/src/components/ProjectSelect.tsx

```diff
@@ -29,39 +30,94 @@ const ProjectSelector: React.FC<Props> = ({
 }) => {
   const { t } = useTranslation();
   const [currentUser] = useCurrentUserInfo();
+  const baiClient = useSuspendedBackendaiClient();
 
   const [value, setValue] = useControllableValue(selectProps);
-  const { projects, user } = useLazyLoadQuery<ProjectSelectorQuery>(
-    graphql`
-      query ProjectSelectorQuery($domain_name: String, $email: String) {
-        projects: groups(domain_name: $domain_name, is_active: true) {
-          id
-          is_active
-          name
-          resource_policy
-        }
-        user(email: $email) {
-          groups {
+  const userRole = useCurrentUserRole();
+  const { projectsSince2403, projectsBefore2403, user } =
+    useLazyLoadQuery<ProjectSelectorQuery>(
+      graphql`
+        query ProjectSelectorQuery(
+          $domain_name: String
+          $email: String
+          $type: [String]
+        ) {
+          projectsSince2403: groups(
+            domain_name: $domain_name
+            is_active: true
+            type: $type
+          ) @since(version: "24.03.0") {
             id
+            is_active
             name
+            resource_policy
+            type
```

Comment:
I have a small question. 

I thought the advantage of grahpql is that you can get only the data you want. But currently, only `id, name, resource_policy` are used among the data fields that received from graphql, is there a reason to explicitly include the unused fields?